internal/locate/region_request.go

```diff
@@ -1040,6 +1044,14 @@ func (s *RegionRequestSender) SendReqCtx(
 			return resp, nil, err
 		}
 
+		var isLocalTraffic bool
+		if staleReadCollector != nil && s.replicaSelector != nil {
+			if target := s.replicaSelector.targetReplica(); target != nil {
+				isLocalTraffic = target.store.IsLabelsMatch(s.replicaSelector.labels)
+				staleReadCollector.onReq(req, isLocalTraffic)
```

Comment:
Should we move the statistic code after `s.sendReqToRegion` finishes with no rpc error?

Reply:
No, since function `staleReadCollector .onResp` will be called after rpc finished.

Reply:
And always need to record request qps and traffic no matter request success.